### PR TITLE
Kestrel

### PR DESCRIPTION
**Pull request by williamboxhall**
Once this is merged, I'll rename the repository to `kestrel` and make it public. Once that is done I can look at publishing to maven central via Sonatype.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,122 @@
...
+*Implements two interfaces, an `EventSink`, for saving events for aggregates, and an `EventSource`, for retrieving those
+events. In general, the Event Store should only ever be written to via the `CommandGateway` and read from via an 
+`EventProcessor`. Kestrel provides support for a postgres backed event store out of the box.*
+- [**EventProcessor**](https://github.com/cultureamp/kotlin-eventsourcing/blob/master/src/main/kotlin/com/cultureamp/eventsourcing/EventProcessor.kt) -
+*Provides an abstraction over any event-processor, for example a `Projector` or a `Reactor`. This takes care of filtering
+out any irrelevant events from being passed to said projectors or reactors.*
+- **Projector** - *An event processor that merely updates a "projection" of the data. Should always be disposable and 
+re-runnable from the beginning of the event-sequence. Should be built in an [idempotent](https://en.wikipedia.org/wiki/Idempotence)
+fashion since event-sourced systems favour asynchronous, distributed systems where it becomes more and more impossible
+to create perfect transactions. Build these as if they are at-least-once delivery of events.*
```
Comment (admackin) [inline README.md:108]:
```suggestion
to create perfect transactions. Build these as if they are an at-least-once delivery of events.*
```

File: src/test/kotlin/com/cultureamp/eventsourcing/example/SurveyAggegate.kt [hunk 2]
```diff
@@ -43,19 +41,12 @@ data class SurveyAggregate(val name: Map<Locale, String>, val accountId: UUID, v
...
 sealed class SurveyCommand : Command
 sealed class SurveyCreationCommand : SurveyCommand(), CreationCommand
-data class CreateSurvey(
-    override val aggregateId: UUID,
-    val surveyCaptureLayoutAggregateId: UUID,
-    val name: Map<Locale, String>,
-    val accountId: UUID,
-    val createdAt: DateTime
-) : SurveyCreationCommand()
+data class CreateSurvey(override val aggregateId: UUID, val surveyCaptureLayoutAggregateId: UUID, val name: Map<Locale, String>, val accountId: UUID, val createdAt: DateTime) : SurveyCreationCommand()
```
Comment (admackin) [inline src/test/kotlin/com/cultureamp/eventsourcing/example/SurveyAggegate.kt:44]: this line is a bit long now

**Review by admackin — CHANGES_REQUESTED** [review]
super-minor cosmetic suggestions

File: src/test/kotlin/com/cultureamp/eventsourcing/example/SurveyAggegate.kt (hunk 2 above)
Comment (williamboxhall) [reply src/test/kotlin/com/cultureamp/eventsourcing/example/SurveyAggegate.kt:44]: you're right, I'll put it back

Comment (williamboxhall) [reply src/test/kotlin/com/cultureamp/eventsourcing/example/SurveyAggegate.kt]: done

**Comment by williamboxhall** [comment]
Force-pushed to squash down @admackin's formatting suggestions and fix a small typo that was introduced.

**Review by admackin — APPROVED** [review]

**Pull request merged**